src/D2L.Bmx/BmxConfigProvider.cs

```diff
@@ -55,6 +55,18 @@ internal class BmxConfigProvider : IBmxConfigProvider {
 	}
 
 	public void SaveConfiguration( BmxConfig config ) {
+
+		string bmxDirectory = BmxPaths.BMX_DIR;
+		string configFileName = BmxPaths.CONFIG_FILE_NAME;
+
+		if( !Directory.Exists( bmxDirectory ) ) {
+			Directory.CreateDirectory( bmxDirectory );
+		}
+
+		if( !File.Exists( configFileName ) ) {
+			File.WriteAllText( configFileName, "" );
+		}
```

Comment:
We need to create the directory, but I don't think this is needed. StreamWriter should create it automatically.
(also using `File.WriteAllText` just to create an empty file feels bad)